### PR TITLE
Fix error swallowing bug

### LINKED ISSUE #109: JavaScript exception being swallowed?
As of v1.3.0, when the js code throws an exception it seems to be getting swallowed by some top-level trap:

```
$ cat examples/exception/script.js

function greet() {
  throw new Error("I am a JS exception")
}
module.exports = { greet };

$ extism call examples/exception.wasm greet --input="Hello" --wasi
Error: Exception generated by QuickJS
returned non-zero exit code: 4294967295
```

This exit code a bit bizarre also. I suspect this might be related to the rquickjs refactor but not sure. When we do figure it out we need to add a test for this.

### PR DESCRIPTION
**Pull request by bhelx**
Fixes #109

Just a test to repro for now. I tried to upgrade quickjs. Did some other house cleaning while in here. It appears that the problem may be in rquickjs, but maybe we are holding it wrong. Will look with Zach today.

### CODE REVIEW

**Comment by zshipko** [comment]
I was finally able to fix this :confetti_ball:  - it came down to using `check_exception` (https://github.com/extism/js-pdk/pull/113/files#diff-3df137e08da293bebdb5cbe1b81fcae6af75992a24c583f47e8040baecbf123aR18) to convert QuickJS errors/exceptions into `anyhow::Error`s 

The commit also includes some fixes for warnings about static mutable values on newer versions of rustc

**Review by zshipko — APPROVED** [review]

**Comment by bhelx** [comment]
Sorry i missed that you got this @zshipko ! gonna try it out

File: Makefile [hunk 1]
```diff
@@ -50,30 +50,40 @@ clean-wasi-sdk:
...
+# 			pip install -r examples/host_funcs/requirements.txt && \
+# 			python examples/host_funcs/host.py examples/host_funcs.wasm && \
+# 			./.venv/Scripts/deactivate.bat
+# else
+# 		@python3 -m venv ./.venv && \
+# 			. ./.venv/bin/activate && \
+# 			pip install -r examples/host_funcs/requirements.txt && \
+# 			python3 examples/host_funcs/host.py examples/host_funcs.wasm && \
+# 			deactivate
+# endif
```
Comment (bhelx) [inline Makefile:65]: I'm gonna leave this uncommented and merge.

Comment (bhelx) [reply Makefile:65]: @chrisdickinson no rush, but could you chime in as to how to fix this or when another sdk release will happen? i don't recall we had a definitive reason why we were still getting the failure to import here.

Comment (chrisdickinson) [reply Makefile:66]: I wasn't able to replicate the `extism_sys` issue on my machine – I ran into a _similar_ issue with `extism-cli` but I solved it by upgrading my virtualenv. I think it _could_ be a virtualenv issue – looking back at the error you posted, it doesn't look like you had the target virtualenv active when you uninstalled `extism_sys` and installed a newer version. The CI looks like it was failing for [a different reason](https://github.com/extism/js-pdk/actions/runs/12164760947/job/33927191400?pr=113#step:9:880), though?

Comment (zshipko) [reply Makefile:66]: This must have been some kind of caching issue, the tests are passing now: https://github.com/extism/js-pdk/pull/116

**Pull request merged**